libs/editor/editor_notes.cpp

```diff
@@ -181,12 +181,24 @@ void Notes::Upload(osm::OsmOAuth const & auth)
 
         LOG(LINFO, ("A note uploaded with id", id));
       }
+      catch (osm::OsmOAuth::NetworkError const & e)
+      {
+        ulock.lock();  // important: restore mutex before returning
```

Comment:
1. Why do you think the lock should be called here?
2. Why lock is not called below?